File: fap_online_backend/fap_online_backend/src/main/java/com/example/demo/dto/WeeklyTimetableDTO.java
package com.example.demo.dto;
import lombok.Builder;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalTime;
@Data @Builder public class WeeklyTimetableDTO {
    private LocalDate scheduleDate;
    private String subjectCode;
    private String roomName;
    private String timeSlot;
    private LocalTime startTime;
    private LocalTime endTime;
    private String status;
}

File: fap_online_ui/lib/models/response/weekly_timetable_dto.dart
class WeeklyTimetableDTO {
  final String scheduleDate;
  final String subjectCode;
  final String roomName;
  final String timeSlot;
  final String startTime;
  final String endTime;
  final String status;

  WeeklyTimetableDTO({
    required this.scheduleDate,
    required this.subjectCode,
    required this.roomName,
    required this.timeSlot,
    required this.startTime,
    required this.endTime,
    required this.status,
  });

  factory WeeklyTimetableDTO.fromJson(Map<String, dynamic> json) {
    return WeeklyTimetableDTO(
      scheduleDate: json['scheduleDate'] ?? '',
      subjectCode: json['subjectCode'] ?? '',
      roomName: json['roomName'] ?? '',
      timeSlot: json['timeSlot'] ?? '',
      startTime: json['startTime'] ?? '',
      endTime: json['endTime'] ?? '',
      status: json['status'] ?? '',
    );
  }
}
